src/main/java/org/broadinstitute/hellbender/engine/FeatureManager.java

```diff
@@ -80,7 +80,7 @@
      * The tool instance containing the FeatureInput argument values that will form the basis of our
      * pool of FeatureDataSources
      */
-    private final CommandLineProgram toolInstance;
+    private transient final CommandLineProgram toolInstance;
```

Comment:
I looked into this more, and it seems that the problem was that Spark was trying to serialize some log4j classes, which are not serializable, and even Kryo can't serialize them. I think there's a fix that makes the logger fields transient and then reinststes them via `readObject` after deserialization. I'll add this to the next PR.